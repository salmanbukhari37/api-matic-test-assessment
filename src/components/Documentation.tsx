import React, { useState } from "react";
import { Box, useMediaQuery, Theme } from "@mui/material";
import PageRenderer from "./PageRenderer";

const Documentation: React.FC = () => {
  const selectedIndex = 0;

  const isSmallScreen = useMediaQuery((theme: Theme) =>
    theme.breakpoints.down("sm")
  );

  return (
    <Box
      display="flex"
      flexDirection={isSmallScreen ? "column" : "row"}
      height="100vh"
    >
      <PageRenderer pageIndex={selectedIndex} />
    </Box>
  );
};

export default Documentation;
