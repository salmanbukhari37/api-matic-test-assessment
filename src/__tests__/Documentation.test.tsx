import { render, screen } from "@testing-library/react";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import { useMediaQuery } from "@mui/material";
import Documentation from "../components/Documentation";

jest.mock(
  "../components/PageRenderer",
  () =>
    ({ pageIndex }: { pageIndex: number }) =>
      <div data-testid="page-renderer">Page Index: {pageIndex}</div>
);

jest.mock("@mui/material", () => {
  const actual = jest.requireActual("@mui/material");
  return {
    ...actual,
    useMediaQuery: jest.fn(),
  };
});

describe("Documentation Component", () => {
  const theme = createTheme();

  test("renders correctly in large screens and matches snapshot", () => {
    (useMediaQuery as jest.Mock).mockReturnValue(false);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <Documentation />
      </ThemeProvider>
    );

    expect(screen.getByTestId("page-renderer")).toHaveTextContent(
      "Page Index: 0"
    );

    expect(screen.getByTestId("page-renderer").parentElement).toHaveStyle(
      "flex-direction: row"
    );

    expect(container).toMatchSnapshot();
  });

  test("renders correctly in small screens and matches snapshot", () => {
    (useMediaQuery as jest.Mock).mockReturnValue(true);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <Documentation />
      </ThemeProvider>
    );

    expect(screen.getByTestId("page-renderer")).toHaveTextContent(
      "Page Index: 0"
    );

    expect(screen.getByTestId("page-renderer").parentElement).toHaveStyle(
      "flex-direction: column"
    );

    expect(container).toMatchSnapshot();
  });
});
