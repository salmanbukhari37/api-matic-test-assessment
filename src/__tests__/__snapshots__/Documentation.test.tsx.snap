// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation Component renders correctly in large screens and matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-126ni31"
  >
    <div
      data-testid="page-renderer"
    >
      Page Index: 
      0
    </div>
  </div>
</div>
`;

exports[`Documentation Component renders correctly in small screens and matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      data-testid="page-renderer"
    >
      Page Index: 
      0
    </div>
  </div>
</div>
`;
